tiatoolbox/models/architecture/__init__.py

```diff
@@ -29,9 +31,22 @@ def fetch_pretrained_weights(model_name: str, save_path: str, overwrite: bool =
         overwrite (bool):
             Overwrite existing downloaded weights.
 
+    Returns:
+        pathlib.Path:
+            The local path to the cached pretrained weights after downloading.
+
     """
+    if model_name not in PRETRAINED_INFO:
+        raise ValueError(f"Pretrained model `{model_name}` does not exist")
+
     info = PRETRAINED_INFO[model_name]
+
+    if save_path is None:
+        file_name = info["url"].split("/")[-1]
+        save_path = os.path.join(rcParam["TIATOOLBOX_HOME"], "models/", file_name)
```

Comment:
rcParam["TIATOOLBOX_HOME"] is now a Path in #625. You can use rcParam["TIATOOLBOX_HOME"] / "models" / file_name